scripts/src/utils/PluginUtils.ts

```diff
@@ -0,0 +1,129 @@
+import { Node, FunctionExpression, Identifier } from 'estree';
+import { generate } from 'escodegen';
+
+function isReference(node, parent) {
+  if (node.type === 'MemberExpression') {
+    return !node.computed && isReference(node.object, node);
+  }
+
+  if (node.type === 'Identifier') {
+    // TODO is this right?
+    if (parent.type === 'MemberExpression') return parent.computed || node === parent.object;
+
+    // disregard the `bar` in { bar: foo }
+    if ((parent.type === 'Property' || parent.type === 'PropertyDefinition') && node !== parent.value) return false;
+
+    // disregard the `bar` in `class Foo { bar () {...} }`
+    if (parent.type === 'MethodDefinition') return false;
+
+    // disregard the `bar` in `export { foo as bar }`
+    if (parent.type === 'ExportSpecifier' && node !== parent.local) return false;
+
+    // disregard the `bar` in `import { bar as foo }`
+    if (parent.type === 'ImportSpecifier' && node === parent.imported) {
+      return false;
+    }
+
+    return true;
+  }
+
+  return false;
+};
+
+function isStatement(type) {
+  const statementTypes = [
+    'BlockStatement',
+    'BreakStatement',
+    'ContinueStatement',
+    'DebuggerStatement',
+    'DoWhileStatement',
+    'EmptyStatement',
+    'ExpressionStatement',
+    'ForInStatement',
+    'ForOfStatement',
+    'ForStatement',
+    'FunctionDeclaration',
+    'IfStatement',
+    'LabeledStatement',
+    'ReturnStatement',
+    'SwitchStatement',
+    'ThrowStatement',
+    'TryStatement',
+    'VariableDeclaration',
+    'WhileStatement',
+    'WithStatement'
+  ];
```

Comment:
statementTypes should be cache or global, otherwise, a new instance is created with each function call